VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

nebutler963 opened the pull request:
Reason for Change:  If the audio track is not available in the stream, then the TSB buffer was getting reset.
Test Procedure: L1, Full stack.
Risks: Low

shripadgithub committed:
Revert "VPLAY-9525: Switching between tune locators not working without explicit stop"

This reverts commit 87c0700957ce70f67fdd194336c80abc06421e58.

shripadgithub committed:
Revert "RDKEMW-5467 : Set PlayerState to IDLE during stop"

This reverts commit 1ee5e66bdaaf8696e5d7533ea4b741164c93455e.

shripadgithub committed:
Revert "VPLAY-9063: AAMP Instance: Send state changed event only after full stop"

This reverts commit ef6208855ddf0d5b25f45ce61c0a8c939fb3d3a9.

pstroffolino committed:
VPLAY-9063 l1 fixes to go along with revert

Reason for Change: revert to avoid blocker; see ticket

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

Reason for Change:  If the audio track is not available in the stream,
then the TSB buffer was getting reset.
Test Procedure: L1, Full stack.
Risks: Low

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e72d3422538786b1a1d843c2fb5b48b8
PRESENT
-->

# Coverity Issue - Check of thread-shared field evades lock acquisition
Thread1 sets "mpStreamAbstractionAAMP" to a new value. Now the two threads have an inconsistent view of "mpStreamAbstractionAAMP" and updates to fields of "mpStreamAbstractionAAMP" or fields correlated with "mpStreamAbstractionAAMP" may be lost.

_High Impact, CWE-543_
_LOCK_EVASION_

## How to fix
 Guard the modification of "mpStreamAbstractionAAMP" and the read used to decide whether to modify "mpStreamAbstractionAAMP" with the same set of locks.

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:4753](undefined//blob/undefined/priv_aamp.cpp#L4753)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
0385e6553fb826b41843f8a1962dbb38
PRESENT
-->

# Coverity Issue - Unchecked return value
Calling "GetCurrentTextTrack" without checking return value (as is done elsewhere 1 out of 1 times).

_Medium Impact, CWE-252_
_CHECKED_RETURN_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:10181](undefined//blob/undefined/priv_aamp.cpp#L10181)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e72d3422538786b1a1d843c2fb5b48b8
PRESENT
-->

# Coverity Issue - Check of thread-shared field evades lock acquisition
Thread1 sets "mpStreamAbstractionAAMP" to a new value. Now the two threads have an inconsistent view of "mpStreamAbstractionAAMP" and updates to fields of "mpStreamAbstractionAAMP" or fields correlated with "mpStreamAbstractionAAMP" may be lost.

_High Impact, CWE-543_
_LOCK_EVASION_

## How to fix
 Guard the modification of "mpStreamAbstractionAAMP" and the read used to decide whether to modify "mpStreamAbstractionAAMP" with the same set of locks.

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:4759](undefined//blob/undefined/priv_aamp.cpp#L4759)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
05e4396568d3c880a08c0668721cc41d
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "this->mAdDuration" without holding lock "PrivateInstanceAAMP.mAdEventQMtx". Elsewhere, "PrivateInstanceAAMP.mAdDuration" is written to with "PrivateInstanceAAMP.mAdEventQMtx" held 1 out of 1 times.

_Medium Impact, CWE-366_
_MISSING_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:2283](undefined//blob/undefined/priv_aamp.cpp#L2283)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
59c199fc71ff333a110953b523743fb8
PRESENT
-->

# Coverity Issue - Unchecked return value
Calling "GetCurrentAudioTrack" without checking return value (as is done elsewhere 2 out of 2 times).

_Medium Impact, CWE-252_
_CHECKED_RETURN_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:10061](undefined//blob/undefined/priv_aamp.cpp#L10061)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e72d3422538786b1a1d843c2fb5b48b8
PRESENT
-->

# Coverity Issue - Check of thread-shared field evades lock acquisition
Thread1 sets "mpStreamAbstractionAAMP" to a new value. Now the two threads have an inconsistent view of "mpStreamAbstractionAAMP" and updates to fields of "mpStreamAbstractionAAMP" or fields correlated with "mpStreamAbstractionAAMP" may be lost.

_High Impact, CWE-543_
_LOCK_EVASION_

## How to fix
 Guard the modification of "mpStreamAbstractionAAMP" and the read used to decide whether to modify "mpStreamAbstractionAAMP" with the same set of locks.

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:4748](undefined//blob/undefined/priv_aamp.cpp#L4748)

nebutler963 committed:
VPLAY-10194 -AAMP Tune() leaking mCdaiObject Object (#249)

VPLAY-10194 -AAMP Tune() leaking mCdaiObject Object

Reason for Change: When tuning mCdaiObject is leaked.
Test Procedure: L1/L2 Full Stack testing
Risks: Low

jameslofthouse-RED committed:
VPLAY-9355 remove catch-up delay when processing a discontinuity (#212)

* VPLAY-9355 Underflow causes aamp to go into pause state
Reason for change: If audio is processing a discontinuity, do not wait for it to catch up to video
Test Procedure: See ticket
Risks: Low

Signed-off-by: James Lofthouse <james_lofthouse@comcast.com>

pstroffolino committed:
rdk-56194 typo fix (#261)

RDK-56194: Integrating ContentProtection Firebolt in AAMP

Reason for Change: spelling fix

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

srikanthreddybijjam-comcast committed:
VPLAY-9989: Fix for AUTO_CAUSES_COPY Coverity Issue  (#206)

VPLAY-9989:Fix for AUTO_CAUSES_COPY Coverity Issue

Reason for change: Resolved Coverity Issue caused by(AUTO_CAUSES_COPY)
Test Procedure:Refer VPLAY-9989 Coverity Issue(AUTO_CAUSES_COPY).
Priority: P1

Signed-off-by: srikanthreddybijjam-comcast<srikanthreddybijjam.2000@gmail.com>

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
43dd6cf2302bfaccb651a06bbaa072a2
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"respData" is passed-by-value as parameter to "std::shared_ptr<_downloadResponse>::shared_ptr(std::shared_ptr<_downloadResponse> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""respData"")" instead of "respData".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:7514](undefined//blob/undefined/priv_aamp.cpp#L7514)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
ceb07891df64ae3b8e64a6dfc7241eb6
PRESENT
-->

# Coverity Issue - Double unlock
"SendEvent" unlocks "this->mEventManager->mMutexVar" while it is unlocked.

_Medium Impact, CWE-765_
_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:8210](undefined//blob/undefined/priv_aamp.cpp#L8210)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
ceb07891df64ae3b8e64a6dfc7241eb6
PRESENT
-->

# Coverity Issue - Double unlock
"SendEvent" unlocks "this->mEventManager->mMutexVar" while it is unlocked.

_Medium Impact, CWE-765_
_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:8229](undefined//blob/undefined/priv_aamp.cpp#L8229)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
6544e5df56acd0c74e03653967055800
PRESENT
-->

# Coverity Issue - Untrusted loop bound
Passing tainted expression "*this->ptr" to "Mp4Demux", which uses it as a loop boundary.

_Medium Impact, CWE-606_
_TAINTED_SCALAR_

## How to fix
 Ensure that tainted values are properly sanitized, by checking that their values are within a permissible range.

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[test/gstTestHarness/gst-test.cpp:197](undefined//blob/undefined/test/gstTestHarness/gst-test.cpp#L197)

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b7e9402d5325cfffffc304da46f07d5d
PRESENT
-->

# Coverity Issue - Double unlock
"PushNextFragment" unlocks "pMediaStreamContext->mutex" while it is unlocked.

_Medium Impact, CWE-765_
_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[fragmentcollector_mpd.cpp:9042](undefined//blob/undefined/fragmentcollector_mpd.cpp#L9042)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e10b3bdb4a48fbf9a8354dcb8918c0cc
PRESENT
-->

# Coverity Issue - Wait without held lock
"FetchFragment" waits while "pMediaStreamContext->mutex" is unlocked.

_Medium Impact, CWE-667_
_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[fragmentcollector_mpd.cpp:5482](undefined//blob/undefined/fragmentcollector_mpd.cpp#L5482)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
0449a0ba518cd72585823907892ca556
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "pMediaStreamContext->numberOfFragmentChunksCached" without holding lock "MediaTrack.mutex". Elsewhere, "MediaTrack.numberOfFragmentChunksCached" is written to with "MediaTrack.mutex" held 3 out of 3 times (2 of these accesses strongly imply that it is necessary).

_Medium Impact, CWE-366_
_MISSING_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[fragmentcollector_mpd.cpp:9101](undefined//blob/undefined/fragmentcollector_mpd.cpp#L9101)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
449fe91ba8e87de05bcf9e0fbc1c94b7
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "this->numberOfFragmentChunksCached" without holding lock "MediaTrack.mutex". Elsewhere, "MediaTrack.numberOfFragmentChunksCached" is written to with "MediaTrack.mutex" held 3 out of 3 times (2 of these accesses strongly imply that it is necessary).

_Medium Impact, CWE-366_
_MISSING_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[streamabstraction.cpp:1840](undefined//blob/undefined/streamabstraction.cpp#L1840)

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP

nebutler963 committed:
VPLAY-9995 TSB is getting reset/cleared on checking/unchecking SAP
